container_images/concourse-metrics/cmd/publish-job-result/main.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	requests "github.com/GoogleCloudPlatform/guest-test-infra/container_images/concourse-metrics/pkg/monitoring"
+)
+
+var (
+	endTimestamp   = flag.Int64("end-timestamp", time.Now().UnixNano()/1000000, "End timestamp of the job run. Defaults to now.")
+	job            = flag.String("job", "", "Concourse job name.")
+	metricPath     = flag.String("metric-path", "", "Path of the custom metric name to use (custom.googleapis.com/[metric-path]).")
+	projectID      = flag.String("project-id", "", "GCP project Id.")
+	pipeline       = flag.String("pipeline", "", "Concourse pipeline name.")
+	task           = flag.String("task", "", "Concourse task name publishing this metric.")
+	resultState    = flag.String("result-state", "", "Concourse job result state ('success' or 'failure')")
+	startTimestamp = flag.Int64("start-timestamp", -1, "Start timestamp of the job run.")
```

Comment:
Good point that this should totally call out the units of the flag, I'll add that. Is there a better way to default the input value to something that will fail validation for an int? Otherwise I can add a comment to show why -1 is set (to purposely fail validation when not provided).

Reply:
i don't know of a way, no. makes sense now that i get why you set it